--- matrix-e2e.c
+++ matrix-e2e.c.new
@@ -949,7 +949,7 @@
                 to_create = (max_keys / 2) - count;
                 need_to_send = TRUE;
             }
-            purple_debug_info("matrixprpl", "%s: %s: %ld\n",
+            purple_debug_info("matrixprpl", "%s: %s: %lld\n",
                            __func__, key_algo, count);
         }
     }
@@ -1520,7 +1520,7 @@
         plaintext[pt_len] = '\0';
         handle_decrypted_olm(pc, conn, cevent_sender, sender_key, plaintext);
     } else {
-        purple_debug_info("matrixprpl", "%s: Type %zd olm\n", __func__, type);
+        purple_debug_info("matrixprpl", "%s: Type %lld olm\n", __func__, type);
     }
     if (plaintext) {
         clear_mem(plaintext, max_plaintext_len);